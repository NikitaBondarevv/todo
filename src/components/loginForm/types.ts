import { IUser } from 'interfaces/IUser'

export type TLoginFormProps = {
  setUser: (user: IUser) => void
}
