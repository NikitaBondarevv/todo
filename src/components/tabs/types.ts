import { ReactNode } from 'react'

export type TTabsProos = {
  selectedIndex: number | string
}

export type TTabsChildren = {
  props: {
    title: string
    children: ReactNode
  }
}

export type TNavigationProps = {
  titles: string[]
  setActiveTab: Function
  activeTabIndex: number
}
