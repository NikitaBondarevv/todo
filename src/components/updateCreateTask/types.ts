export type TUpdateCreateTaskProps = {
  getTasks: Function
  text: string
  activeTabIndex: number
}

export type TTarget = {
  target: {
    value: string
  }
}
