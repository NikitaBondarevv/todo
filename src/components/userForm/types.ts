import { IUser } from 'interfaces/IUser'

export type TUserForm = {
  disabledFields: string[]
  user: IUser
  onSubmit: (user: IUser) => Promise<void>
}

export type TTarget = {
  target: {
    value: string
    name: string
  }
}

export type TField = {
  label: string
  reg: RegExp
  secure?: boolean
}

type TRegisterField = Record<string, string>

export type TRegisterFields = Record<string, TRegisterField>
