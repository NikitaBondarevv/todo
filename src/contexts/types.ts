import { IUser } from 'interfaces/IUser'

export type TUser = {
  isAuthenticated?: boolean,
  user?: IUser,
  setUser: (user?: IUser) => void
}
