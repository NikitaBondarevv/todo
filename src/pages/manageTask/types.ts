import { IData } from 'interfaces/IData'

export type TTaskProps = {
  onSubmit: (data: IData) => Promise<void>
  text?: string
}

export type TTarget = {
  target: {
    value: string
  }
}
