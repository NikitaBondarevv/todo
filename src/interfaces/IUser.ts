export interface IUser {
  firstName: string
  lastname: string
  email: string
  password: string
  id?: string
  error?: string
}
