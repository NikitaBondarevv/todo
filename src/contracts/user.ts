import { request } from './request'
import { IData } from 'interfaces/IData'

export const createUser = async (data: IData) => request.post('public/user', data)
export const updateUser = async (data: IData) => request.put('user', data)
