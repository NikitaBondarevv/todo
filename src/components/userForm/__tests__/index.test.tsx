import { render, screen } from '@testing-library/react'
import { IUser } from 'interfaces/IUser'

import { user } from '__mocks__/entities/user.mock'
import { UserForm } from '..'
import { TUserForm } from '../types'

describe('<UserForm />', () => {
  const defaultProps: TUserForm = {
    disabledFields: [],
    user,
    onSubmit: jest.fn()
  }
  test('should match snapshot', () => {
    const { asFragment } = render(<UserForm {...defaultProps} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should enable submit button if all fields are filled', () => {
    const user = {
      ...defaultProps.user,
      passwordRepeat: defaultProps.user?.password
    } as IUser
    render(<UserForm {...defaultProps} user={user} />)

    expect(screen.getByDisplayValue('SAVE').getAttribute('disabled')).toBe(null)
  })
})