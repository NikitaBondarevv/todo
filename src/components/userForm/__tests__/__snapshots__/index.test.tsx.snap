// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserForm /> should match snapshot 1`] = `
<DocumentFragment>
  <form
    class="loginForm"
  >
    <ul>
      <li
        class="email"
      >
        <input
          class="success"
          name="email"
          placeholder="EMAIL"
          type="text"
          value="test@gmail.com"
        />
      </li>
      <li
        class="firstName"
      >
        <input
          class="success"
          name="firstName"
          placeholder="FIRST NAME"
          type="text"
          value="firstName"
        />
      </li>
      <li
        class="lastName"
      >
        <input
          class="success"
          name="lastName"
          placeholder="LAST NAME"
          type="text"
          value="lastName"
        />
      </li>
      <li
        class="password"
      >
        <input
          class="success"
          name="password"
          placeholder="PASSWORD"
          type="password"
          value="test-password"
        />
      </li>
      <li
        class="passwordRepeat"
      >
        <input
          class="success"
          name="passwordRepeat"
          placeholder="PASSWORD REPEAT"
          type="password"
          value=""
        />
      </li>
    </ul>
    <input
      disabled=""
      type="submit"
      value="SAVE"
    />
  </form>
</DocumentFragment>
`;
